models/io/HyperClient.cfc

```diff
@@ -530,6 +530,27 @@ component accessors="true" threadSafe singleton {
 			.json();
 	}
 
+	/**
+	 * Request a vector of terms for the given index, document or document ID, and field names
+	 *
+	 * @indexName 	string		Index name or alias.
+	 * @params		struct		Struct of query parameters to influence the request. For example: `"offsets": false }`
+	 * @body		struct		Body payload to send. For example: `{ "filter": { "max_num_terms": 3 } }`
+	 */
+	struct function getTermVectors( required string indexName, string id = "", struct params = {}, struct body = {} ){
```

Comment:
Let’s add a method to the SearchBuilder that passes through to this

Reply:
In the SearchBuilder method, I think we can be more verbose on the arguments to make it more fluent

Reply:
How about this?

```js
struct function getTermVectors( string id = "", string fields = "", struct options = {} ){
	var args = {
		indexName = variables.index,
		id        = arguments.id,
		params    = arguments.options,
		body      = {}
	};
	if ( arguments.options.keyExists( "doc" ) ){
		args.body[ "doc" ] = arguments.options.doc;
		arguments.options.delete( "doc" );
	}
	if ( arguments.options.keyExists( "filter" ) ){
		args.body[ "filter" ] = arguments.options.filter;
		arguments.options.delete( "filter" );
	}
	return getClient().getTermVectors( argumentCollection = args );
}
```

This lets you use the following shorthand:

```js
var result = getInstance( "SearchBuilder@cbElasticsearch" )
                .new( "books" )
                .getTermVectors(
                    myDocument._id,
                    "title,author.name"
                );
```

or longhand, if you have a penchant for pain:

```js
var result = getInstance( "SearchBuilder@cbElasticsearch" )
                .new( "books" )
                .getTermVectors(
                    myDocument._id,
                    "title,author.name",
                    {
                        "field_statistics" : false,
                        "payloads" : false,
                        "filter" : {
                            "min_term_freq": 1,
                            "min_word_length" : "4"
                        }
                    }
                );
```

Reply:
Great, it's already committed. :)

Reply:
Add `fields` argument cleanup to accept an array.